include/tilck/common/string_util.h

```diff
@@ -29,14 +29,28 @@
 
 #endif
 
-int strcmp(const char *s1, const char *s2);
-int strncmp(const char *s1, const char *s2, size_t n);
-int memcmp(const void *m1, const void *m2, size_t n);
-char *strstr(const char *haystack, const char *needle);
-char *strcpy(char *dest, const char *src);
-char *strncpy(char *dest, const char *src, size_t n);
-char *strcat(char *dest, const char *src);
-char *strncat(char *dest, const char *src, size_t n);
+#if !defined(TESTING) && !defined(USERMODE_APP)
```

Comment:
You're already inside of a `#if !defined(TESTING) && !defined(USERMODE_APP)` block.
It doesn't make any sense to add another such `#if` here. Please, check better the context on your changes.

Reply:
Ok, sorry

Reply:
Btw, i think always indenting when inside of an ifdef can prevent this problem in the future. I think you considered that already, but i just wanted to point it out